### PR TITLE
Fixed typing exception throwing issues with JIT

### LINKED ISSUE #3027: #2860 regressed torch.jit.script(densenet161).save
## 🐛 Bug

```
Traceback (most recent call last):
  File "repro.py", line 7, in <module>
    torch.jit.script(model).save('densenet161.pt')
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_script.py", line 911, in script
    return torch.jit._recursive.create_script_module(
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 370, in create_script_module
    return create_script_module_impl(nn_module, concrete_type, stubs_fn)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 426, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_script.py", line 388, in _construct
    init_fn(script_module)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 406, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 426, in create_script_module_impl
    script_module = torch.jit.RecursiveScriptModule._construct(cpp_module, init_fn)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_script.py", line 388, in _construct
    init_fn(script_module)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 406, in init_fn
    scripted = create_script_module_impl(orig_value, sub_concrete_type, stubs_fn)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 382, in create_script_module_impl
    method_stubs = stubs_fn(nn_module)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 618, in infer_methods_to_compile
    stubs.append(make_stub_from_method(nn_module, method))
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 52, in make_stub_from_method
    return make_stub(func, method_name)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/_recursive.py", line 37, in make_stub
    ast = get_jit_def(func, name, self_name="RecursiveScriptModule")
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/frontend.py", line 259, in get_jit_def
    return build_def(ctx, fn_def, type_line, def_name, self_name=self_name)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/torch/jit/frontend.py", line 288, in build_def
    type_comment_decl = torch._C.parse_type_comment(type_line)
RuntimeError: expected type comment but found 'def' here:
def forward(self, init_features: Tensor) -> Tensor:  # type: ignore[override]
~~~ <--- HERE
```

## To Reproduce

1. Install nightly build `pip install --upgrade --pre torchvision -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html`
2. Run:
```
import torch
import torchvision
model = torchvision.models.densenet161(pretrained=False)
model.eval()
torch.jit.script(model).save('densenet161.pt')
```

## Environment

Collecting environment information...
PyTorch version: 1.8.0.dev20201118
Is debug build: False
OS: Mac OSX 11.0.1 (x86_64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.27)
CMake version: version 3.18.4
Python version: 3.8 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Versions of relevant libraries:
[pip] numpy==1.19.0
[pip] torch==1.8.0.dev20201118
[pip] torchtext==0.7.0
[pip] torchvision==0.9.0.dev20201118
[conda] Could not collect

## Additional context

@frgfm @fmassa @pmeier #2860

### PR DESCRIPTION
**Pull request by frgfm**
As per #3027, some JIT issues were introduced in #2860 with typing, that weren't caught initially by tests.
This PR addresses the issues related to the densenet module.

This PR should close #3027

### CODE REVIEW

**Comment by frgfm** [comment]
I'm still getting the following error with mypy:
```
torchvision/models/densenet.py:74: error: Name 'forward' already defined on line 70  [no-redef]
        @torch.jit._overload_method  # noqa: F811
         ^
torchvision/models/densenet.py:80: error: Name 'forward' already defined on line 70  [no-redef]
        def forward(self, input: Tensor) -> Tensor:  # noqa: F811
        ^
```

But I'm not very knowledgeable on JIT overlead decorator. Should we accept that some mypy tests fail? Or is there a way around this without having JIT issues? @fmassa @pmeier

**Comment by pmeier** [comment]
> Should we accept that some mypy tests fail?

Sort of. I suggest we simply exclude `densenet.py` from the type check for now and only re-enable it if the issue with torchscript is fixed.

File: torchvision/models/densenet.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def __init__(
             )
             self.add_module('denselayer%d' % (i + 1), layer)
 
-    def forward(self, init_features: Tensor) -> Tensor:  # type: ignore[override]
+    def forward(self, init_features):
```
Comment (pmeier) [inline torchvision/models/densenet.py:124]: Only the `type: ignore` comment has to go.

```suggestion
    def forward(self, init_features: Tensor) -> Tensor:
```

Comment (frgfm) [reply torchvision/models/densenet.py]: Fixed!

**Comment by frgfm** [comment]
Alright, it should be good to go now :ok_hand:

**Review by pmeier — APPROVED** [review]

**Review by fmassa — APPROVED** [review]
Thanks!

**Comment by pmeier** [comment]
@frgfm Do you want to have another go at this as https://github.com/pytorch/pytorch/pull/51675 fixed the behavior of `type: ignore` comments for `torch.jit`?

**Comment by frgfm** [comment]
@pmeier Thanks for the mention! Yes, it would be bad karma for me to leave this unfinished haha

**Comment by frgfm** [comment]
@pmeier Here it is https://github.com/pytorch/vision/pull/3395 !

**Pull request merged**